### PR TITLE
[DST-387]: Generate property tables in our documentation from code

### PR DESCRIPTION
**Pull request by OsamaAbdellateef**
Making the props dynamic based on the the types inside the component

### CODE REVIEW

File: docs/scripts/build-props-table.mjs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  // include asserted types for components eg. Autocomplete & Select
+  customComponentTypes: ['AutocompleteComponent', 'SelectComponent'],
+});
+
+// Resolve __dirname for ESM
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+const componentsDir = path.resolve(__dirname, '../../packages/components/src');
+const outputFilePath = path.resolve(__dirname, '../.table-props/index.json');
```
Comment (sebald) [inline docs/scripts/build-props-table.mjs:23]: I find table props somewhat confusing. Rather call it "component props"?, maybe we can put it next to the registry file? 

The file should be added to gitignore

File: docs/scripts/build-props-table.mjs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+fs.ensureDirSync(path.dirname(outputFilePath));
+
+const generatePropsTables = () => {
+  // Getting all directories
+  const componentDirs = fs
```
Comment (sebald) [inline docs/scripts/build-props-table.mjs:29]: Use globby or minmatch here so you don't have to do it by hand?

File: docs/scripts/build-props-table.mjs [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import fs from 'fs-extra';
```
Comment (sebald) [inline docs/scripts/build-props-table.mjs:1]: This should be zx script, so you have access to all relevant libs (fs-extra, globby, etc)

**Review by sebald — CHANGES_REQUESTED** [review]

File: docs/scripts/build-props-table.mjs (hunk 1 above)
Comment (OsamaAbdellateef) [reply docs/scripts/build-props-table.mjs:23]: :heavy_check_mark: 
didn't get "we put it next to reg file "

File: docs/scripts/build-props-table.mjs (hunk 3 above)
Comment (OsamaAbdellateef) [reply docs/scripts/build-props-table.mjs:1]: zx doesn't include fs-extra :( @sebald

File: docs/package.json [hunk 4]
```diff
@@ -24,10 +25,12 @@
     "contentlayer2": "0.4.6",
     "eslint": "8.57.0",
     "eslint-config-next": "14.1.4",
+    "fs-extra": "11.2.0",
```
Comment (sebald) [inline docs/package.json:28]: Remove? :)

File: docs/scripts/build-component-props.mjs [hunk 5]
```diff
@@ -0,0 +1,57 @@
+import fs from 'fs-extra';
```
Comment (sebald) [inline docs/scripts/build-component-props.mjs:1]: This is also part of zx

File: docs/scripts/build-component-props.mjs [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+import path from 'path';
+import { dirname } from 'path';
```
Comment (sebald) [inline docs/scripts/build-component-props.mjs:3]: Imported twice. Also when importing node modules, we should use node:path for security :)

**Review by sebald — CHANGES_REQUESTED** [review]

**Comment by sebald** [comment]
Sorry, some small adjustments 🙈

File: docs/scripts/build-props-table.mjs (hunk 3 above)
Comment (sebald) [reply docs/scripts/build-props-table.mjs]: https://google.github.io/zx/api#fs :P

Comment (OsamaAbdellateef) [reply docs/scripts/build-props-table.mjs]: aaaaaaaaaaaah mb ✅

**Review by sebald — APPROVED** [review]

**Pull request merged**